### PR TITLE
More memory stat.

### PR DESCRIPTION
**Pull request by AlexK0**
This patch adds a bunch of different memory stats.

WIP while #160 is not merged.

### CODE REVIEW

File: common/dl-utils-lite.h [hunk 1]
```diff
@@ -42,24 +42,24 @@ void dl_assert__ (const char *expr, const char *file_name, const char *func_name
...
-typedef struct {
-  unsigned long long vm_peak;
-  unsigned long long vm;
-  unsigned long long rss_peak;
-  unsigned long long rss;
-  unsigned long long rss_file;
-  unsigned long long rss_shmem;
-} mem_info_t;
+struct mem_info_t {
+  uint32_t vm_peak;
```
Comment (eugene536) [inline common/dl-utils-lite.h:54]: will `uint32_t` be enough?

File: server/php-worker-stats.h [hunk 2]
```diff
@@ -57,7 +64,24 @@ class PhpWorkerStats {
...
+
+  std::array<int64_t, PERCENTILE_SAMPLES> workers_rss_usage_{};
+  std::array<int64_t, PERCENTILE_SAMPLES> workers_vms_usage_{};
+  std::array<int64_t, PERCENTILE_SAMPLES> workers_shm_usage_{};
+
+  uint64_t rss_kb_sum_{0};
+  uint64_t vms_kb_sum_{0};
+
+  uint32_t rss_kb_max_{0};
+  uint32_t vms_kb_max_{0};
```
Comment (eugene536) [inline server/php-worker-stats.h:82]: Why do we use `uint64_t` for `rss_kb_sum_` but `uint32_t` for `rss_kb_max`?

File: common/dl-utils-lite.h (hunk 1 above)
Comment (AlexK0) [reply common/dl-utils-lite.h:54]: Yep, they are in kbytes.

File: server/php-worker-stats.h (hunk 2 above)
Comment (AlexK0) [reply server/php-worker-stats.h:82]: Yeah..

rss_kb_max - in kbytes, so uint32_t is enough here
rss_kb_sum_ - sum for all processes in kbytes, and I'm not sure if uint32_t is enough for sum, but uint64_t is enough
workers_rss_usage_ - in bytes..

I'm changing rss_kb_sum_ to int64, to be more consistent.

**Review by DrDet — APPROVED** [review]

File: server/php-master.cpp [hunk 3]
```diff
@@ -1493,20 +1467,23 @@ class QPSCalculator {
   }
 
   void update(double time_point, const PhpWorkerStats &new_worker_stats) {
-    if (!prev_worker_stats_.total_queries()) {
-      prev_worker_stats_ = new_worker_stats;
+    if (!prev_total_queries_) {
```
Comment (unserialize) [inline server/php-master.cpp:1470]: I can't find, when this field is set to 0. I mean, while execution, when multiple requests are processed one by one

Comment (AlexK0) [reply server/php-master.cpp:1470]: It is set to 0 only on this struct initialization (1502 line), and then it is only incrementally updated.

Comment (unserialize) [reply server/php-master.cpp:1470]: It's not annulated between requests. Is it on purpose?

Comment (AlexK0) [reply server/php-master.cpp:1470]: Yes, it is ok. new_worker_stats.total_queries() returns the monotonically incremented requests count, so we get a delta between the old value and the new one.

**Review by unserialize — APPROVED** [review]

**Pull request merged**